src/rgw/rgw_iam_policy.cc

```diff
@@ -202,12 +205,9 @@ ARN::ARN(const rgw_bucket& b, const string& o)
 
 optional<ARN> ARN::parse(const string& s, bool wildcards) {
   static const regex rx_wild("arn:([^:]*):([^:]*):([^:]*):([^:]*):([^:]*)",
-			     std::regex_constants::ECMAScript |
-			     std::regex_constants::optimize);
+			     ECMAScript | optimize);
```

Comment:
```
/home/jenkins-build/build/workspace/ceph-pull-requests/src/rgw/rgw_iam_policy.cc:210:78: error: call of overloaded ‘basic_regex(const char [49], int)’ is ambiguous
     "arn:([^:*]*):([^:*]*):([^:*]*):([^:*]*):([^:*]*)", ECMAScript | optimize);
                                                                              ^
In file included from /home/jenkins-build/build/workspace/ceph-pull-requests/build/boost/include/boost/regex/v4/regex.hpp:67:0,
                 from /home/jenkins-build/build/workspace/ceph-pull-requests/build/boost/include/boost/regex.hpp:31,
                 from /home/jenkins-build/build/workspace/ceph-pull-requests/src/rgw/rgw_iam_policy.cc:10:
/home/jenkins-build/build/workspace/ceph-pull-requests/build/boost/include/boost/regex/v4/basic_regex.hpp:337:4: note: candidate: boost::basic_regex<charT, traits>::basic_regex(const charT*, const charT*, boost::basic_regex<charT, traits>::flag_type) [with charT = char; traits = boost::regex_traits<char>; boost::basic_regex<charT, traits>::flag_type = unsigned int]
    basic_regex(const charT* p1, const charT* p2, flag_type f = regex_constants::normal)
    ^
/home/jenkins-build/build/workspace/ceph-pull-requests/build/boost/include/boost/regex/v4/basic_regex.hpp:333:13: note: candidate: boost::basic_regex<charT, traits>::basic_regex(const charT*, boost::basic_regex<charT, traits>::flag_type) [with charT = char; traits = boost::regex_traits<char>; boost::basic_regex<charT, traits>::flag_type = unsigned int]
    explicit basic_regex(const charT* p, flag_type f = regex_constants::normal)
             ^
/home/jenkins-build/build/workspace/ceph-pull-requests/src/rgw/rgw_iam_policy.cc: In function ‘boost::optional<rgw::auth::Principal> rgw::IAM::parse_principal(CephContext*, rgw::IAM::TokenID, std::__cxx11::string&&)’:
```

i guess, the type of OR'ed enums can be taken as `unsigned int` or `const char*` by GCC 4:5.3.1-1ubuntu1.  we might need to cast it to `regex::flag_type` explicitly.